### PR TITLE
[feat] Zeroize the crypto hardware blocks

### PR DESCRIPTION
**Pull request by mhatrevi**
This change updates the crypto drivers to zeroize a crypto block
on completion of an crypto operation. Additionally, in the event of an error condition,
the ROM has been updated to zeroize all the crypto hardware blocks.

### CODE REVIEW

**Review by JohnTraverAmd — APPROVED** [review]

File: drivers/src/ecc384.rs [hunk 1]
```diff
@@ -286,11 +290,16 @@ impl Ecc384 {
...
+        let result = verify_r == signature.r;
+
+        self.zeroize();
+
+        Ok(result)
+    }
+
+    /// Zeroize the hardware registers.
+    pub fn zeroize(&mut self) {
+        self.ecc.regs_mut().ctrl().write(|w| w.zeroize(true));
```
Comment (korran) [inline drivers/src/ecc384.rs:302]: Does the zeroize operation really happen in a single cycle? Do we need to wait for this bit to clear before we can use the ECC peripheral again?

Comment (korran) [reply drivers/src/ecc384.rs:302]: As best as I can tell from the RTL, this doesn't seem to be a state machine.

Comment (mhatrevi) [reply drivers/src/ecc384.rs:302]: It's a one cycle operation; this has been confirmed with the hardware team.

**Review by korran — APPROVED** [review]

File: rom/dev/src/main.rs [hunk 2]
```diff
@@ -115,7 +115,8 @@ extern "C" fn exception_handler(exception: &exception::ExceptionRecord) {
 
     // TODO: Signal non-fatal error to SOC
 
-    report_error(CaliptraError::ROM_GLOBAL_EXCEPTION.into());
+    let mut env = unsafe { rom_env::RomEnv::new_from_registers() };
```
Comment (korran) [inline rom/dev/src/main.rs:118]: We shouldn't call this here, as some drivers (in particular, CSRNG), will need to do some initialization, which could cause some serious problems in a trap.

These zeroization calls are probably going to need to be unsafe static functions, similar to `Mailbox::abort_pending_soc_to_uc_transactions()`.

**Pull request merged**